components/org.wso2.carbon.identity.authenticator.risk.analytics.endpoint/src/test/java/org/wso2/carbon/identity/authenticator/risk/analytics/endpoint/util/RiskScoreServiceUtilTest.java

```diff
@@ -0,0 +1,245 @@
+            Assert.assertEquals(e.getMessage(), "Invalid config element with no username in " +
+                    Constants.RISK_CALCULATOR_CONFIG_XML);
+        }
+        appendNode(Constants.USERNAME, "admin");
+    }
+
+    @Test
+    public void testMissingPassword() throws Exception {
+        System.setProperty(ServerConstants.CARBON_CONFIG_DIR_PATH, pathToConfigFileLocation);
+        deleteNode(Constants.PASSWORD);
+        try {
+            RiskScoreServiceUtil.loadServerConfig();
+        } catch (RiskScoreServiceConfigurationException e) {
+            Assert.assertNotNull(e);
+            Assert.assertEquals(e.getMessage(), "Invalid config element with no password in " +
+                    Constants.RISK_CALCULATOR_CONFIG_XML);
+        }
+        appendNode(Constants.PASSWORD, "admin");
+    }
+
+    @Test
+    public void testMissingAuthStream() throws Exception {
+        System.setProperty(ServerConstants.CARBON_CONFIG_DIR_PATH, pathToConfigFileLocation);
+        deleteNode(Constants.AUTHENTICATION_STREAM);
+        try {
+            RiskScoreServiceUtil.loadServerConfig();
+        } catch (RiskScoreServiceConfigurationException e) {
+            Assert.assertNotNull(e);
+            Assert.assertEquals(e.getMessage(), "Invalid config element with no authentication stream in " +
+                    Constants.RISK_CALCULATOR_CONFIG_XML);
+        }
+        appendNode(Constants.AUTHENTICATION_STREAM, "org.wso2.is.analytics.stream.RiskScoreRequest:1.0.0");
+    }
+
+    @Test
+    public void testMissingRiskScoreStream() throws Exception {
+        System.setProperty(ServerConstants.CARBON_CONFIG_DIR_PATH, pathToConfigFileLocation);
+        deleteNode(Constants.RISK_SCORE_STREAM);
+        try {
+            RiskScoreServiceUtil.loadServerConfig();
+        } catch (RiskScoreServiceConfigurationException e) {
+            Assert.assertNotNull(e);
+            Assert.assertEquals(e.getMessage(), "Invalid config element with no risk score stream in " +
+                    Constants.RISK_CALCULATOR_CONFIG_XML);
+        }
+        appendNode(Constants.RISK_SCORE_STREAM, "org.wso2.is.analytics.stream.RiskScorePerRule:1.0.0");
+    }
+
+
+    @Test(expectedExceptions = RiskScoreServiceConfigurationException.class)
+    public void testFileNotFoundException() throws RiskScoreServiceConfigurationException {
+        System.setProperty(ServerConstants.CARBON_CONFIG_DIR_PATH, "null");
+        try {
+            RiskScoreServiceUtil.loadServerConfig();
+        } catch (RiskScoreServiceConfigurationException e) {
+            Assert.assertEquals(e.getCause().getClass(), FileNotFoundException.class);
+            throw e;
+        }
+    }
+
+//    @Test
+//    public void testXMLException() throws Exception {
+//        System.setProperty(ServerConstants.CARBON_CONFIG_DIR_PATH, pathToConfigFileLocation);
+//
+//        appendNode("error", "</error>");
+//        RiskScoreServiceUtil.loadServerConfig();
+//
+//    }
```

Comment:
Is this test needed? or should it be removed?

Reply:
removed